libsmack/libsmack.c

```diff
@@ -576,14 +577,14 @@ const char *smack_smackfs_path(void)
 	return smackfs_mnt;
 }
 
-ssize_t smack_new_label_from_self(char **label)
+static ssize_t smack_new_label_from_proc(const char *proc_path, char **label)
 {
 	char buf[SMACK_LABEL_LEN + 1];
 	char *result;
 	int fd;
 	int ret;
```

Comment:
"read" returns a ssize_t .  Could you profit of this evolution to avoid the ssize_t -> int cast?